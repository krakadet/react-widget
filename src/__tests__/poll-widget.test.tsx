import React from 'react';
import '@testing-library/jest-dom'
import { render, fireEvent, screen } from "@testing-library/react";
import PollWidget from '../component/poll-widget';
import { setLocalStorage, getLocalStorage } from '../utils';


jest.mock('../utils' , () => ({
  setLocalStorage: jest.fn(),
  getLocalStorage: jest.fn()
}));

describe('Test poll widdet component', ()=> {

    const mockProps = {
        mainQuestion: 'What is your favorite color?',
        answers: ['Red', 'Blue', 'Green'],
    }

    it('should render initial state with zeros', () => {
        // @ts-ignore
        getLocalStorage.mockImplementation(() => [0, 0, 0]);
        const {container} = render(<PollWidget {...mockProps} />)
        expect(container).toMatchSnapshot()
    });

    it('should render initial state with different voutes', () => {
        // @ts-ignore
        getLocalStorage.mockImplementation(() => [25, 35, 45]);
        const {container, findAllByText, getByText} = render(<PollWidget {...mockProps} />)
        expect(getByText(/25%/i)).toBeInTheDocument()
        expect(getByText(/35%/i)).toBeInTheDocument()
        expect(getByText(/45%/i)).toBeInTheDocument()
    });

    it('should update localStorageVoutes', () => {
        // @ts-ignore
        getLocalStorage.mockImplementation(() => [25, 35, 45]);
        const {container, findAllByText, getByText} = render(<PollWidget {...mockProps} />)
        const firstAnswer = screen.getByText(/25%/i)
        fireEvent.click(firstAnswer)
        expect(setLocalStorage).toHaveBeenCalledWith(mockProps.mainQuestion, [26, 35, 45])
        console.log(screen.debug(container))
        expect(getByText(/26%/i)).toBeInTheDocument()
    });
});