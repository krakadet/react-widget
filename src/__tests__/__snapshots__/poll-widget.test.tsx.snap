// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test poll widdet component should render initial state with zeros 1`] = `
<div>
  <div
    class="container"
  >
    <h1>
      What is your favorite color?
    </h1>
    <div
      class="progressbar-container"
    >
      <div
        class="progressbar-complete"
        style="width: 0%;"
      >
        <div
          class="progressbar-liquid"
        />
      </div>
      <span
        class="progress"
      >
        Red
         
        0
        %
      </span>
    </div>
    <div
      class="progressbar-container"
    >
      <div
        class="progressbar-complete"
        style="width: 0%;"
      >
        <div
          class="progressbar-liquid"
        />
      </div>
      <span
        class="progress"
      >
        Blue
         
        0
        %
      </span>
    </div>
    <div
      class="progressbar-container"
    >
      <div
        class="progressbar-complete"
        style="width: 0%;"
      >
        <div
          class="progressbar-liquid"
        />
      </div>
      <span
        class="progress"
      >
        Green
         
        0
        %
      </span>
    </div>
  </div>
</div>
`;
